wagtail/models/revisions.py

```diff
@@ -266,19 +288,15 @@ class RevisionMixin(models.Model):
     @property
     def revisions(self):
         """
-        Returns revisions that belong to the object.
-
-        Subclasses should define a
-        :class:`~django.contrib.contenttypes.fields.GenericRelation` to
-        :class:`~wagtail.models.Revision` and override this property to return
-        that ``GenericRelation``. This allows subclasses to customize the
-        ``related_query_name`` of the ``GenericRelation`` and add custom logic
-        (e.g. to always use the specific instance in ``Page``).
+        Returns revisions that belong to the object. For non-page models, this
+        is done by querying the :class:`~wagtail.models.Revision` model directly
+        rather than using a
+        :class:`~django.contrib.contenttypes.fields.GenericRelation`, to avoid
+        `a known limitation <https://code.djangoproject.com/ticket/31269>`_ in
+        Django for models with multi-table inheritance where the relation's
+        content type may not match the instance's type.
         """
-        return Revision.objects.filter(
-            content_type=self.get_content_type(),
-            object_id=self.pk,
-        )
+        return Revision.objects.for_instance(self)
```

Comment:
This was done in the original PR as [`da8b572` (#11750)](https://github.com/wagtail/wagtail/pull/11750/commits/da8b57206988ae1dfa770271a36c2f57b8f1796b), but the fix in `for_instance` itself has now been done in https://github.com/wagtail/wagtail/commit/1b691a905f5baa0b6fceba8e313ca4ec78ad7ecf.